### ISSUE TITLE
Cuckoo rooter issue

### ISSUE THREAD

**Issue by manzy90** [cuckoosandbox/cuckoo#2514]
Hi my cuckoo was working fine then when i turned it off then back on i am getting this error everytime ti try to start it.

 Cuckoo Sandbox 2.0.6
 www.cuckoosandbox.org
 Copyright (c) 2010-2018

2018-10-05 22:45:10,683 [cuckoo] CRITICAL: CuckooStartupError: The rooter is required but we can't connect to it as the rooter is not actually running. Please refer to the documentation on working with the rooter.

Any help please?

**Comment by manzy90** [comment]
cuckoo.conf
[cuckoo]
# Enable or disable startup version check. When enabled, Cuckoo will connect
# to a remote location to verify whether the running version is the latest
# one available.
version_check = off

# If turned on, Cuckoo will delete the original file after its analysis
# has been completed.
delete_original = off

# If turned on, Cuckoo will delete the copy of the original file in the
# local binaries repository after the analysis has finished. (On *nix this
# will also invalidate the file called "binary" in each analysis directory,
# as this is a symlink.)
delete_bin_copy = off

# Specify the name of the machinery module to use, this module will
# define the interaction between Cuckoo and your virtualization software
# of choice.
machinery = virtualbox

# Enable creation of memory dump of the analysis machine before shutting
# down. Even if turned off, this functionality can also be enabled at
# submission. Currently available for: VirtualBox and libvirt modules (KVM).
memory_dump = off

# When the timeout of an analysis is hit, the VM is just killed by default.
# For some long-running setups it might be interesting to terminate the
# monitored processes before killing the VM so that connections are closed.
terminate_processes = off

# Enable automatically re-schedule of "broken" tasks each startup.
# Each task found in status "processing" is re-queued for analysis.
reschedule = off

# Enable processing of results within the main cuckoo process.
# This is the default behavior but can be switched off for setups that
# require high stability and process the results in a separate task.
process_results = on

# Limit the amount of analysis jobs a Cuckoo process goes through.
# This can be used together with a watchdog to mitigate risk of memory leaks.
max_analysis_count = 0

# Limit the number of concurrently executing analysis machines.
# This may be useful on systems with limited resources.
# Set to 0 to disable any limits.
max_machines_count = 0

# Limit the amount of VMs that are allowed to start in parallel. Generally
# speaking starting the VMs is one of the more CPU intensive parts of the
# actual analysis. This option tries to avoid maxing out the CPU completely.
max_vmstartup_count = 10

# Minimum amount of free space (in MB) available before starting a new task.
# This tries to avoid failing an analysis because the reports can't be written
# due out-of-diskspace errors. Setting this value to 0 disables the check.
# (Note: this feature is currently not supported under Windows.)
freespace = 64

# Temporary directory containing the files uploaded through Cuckoo interfaces
# (api.py and Django web interface). Defaults to the default temporary
# directory of the operating system (e.g., /tmp on Linux). Overwrite the value
# if you'd like to specify an alternative path.
tmppath = /tmp

# Path to the unix socket for running root commands.
rooter = {{ cuckoo.cuckoo.rooter }}

[feedback]
# Cuckoo is capable of sending "developer feedback" to the developers so that
# they can more easily improve the project. This functionality also allows the
# user to quickly request new features, report bugs, and get in touch with
# support in general, etc.
enabled = off
name = {{ cuckoo.feedback.name }}
company = {{ cuckoo.feedback.company }}
email = {{ cuckoo.feedback.email }}

[resultserver]
# The Result Server is used to receive in real time the behavioral logs
# produced by the analyzer.
# Specify the IP address of the host. The analysis machines should be able
# to contact the host through such address, so make sure it's valid.
# NOTE: if you set resultserver IP to 0.0.0.0 you have to set the option
# `resultserver_ip` for all your virtual machines in machinery configuration.
ip = 192.168.56.1

# Specify a port number to bind the result server on.
port = 2042

# Force the port chosen above, don't try another one (we can select another
# port dynamically if we can not bind this one, but that is not an option
# in some setups)
force_port = off

# Maximum size of uploaded files from VM (screenshots, dropped files, log).
# The value is expressed in bytes, by default 128 MB.
upload_max_size = 10485760

[processing]
# Set the maximum size of analyses generated files to process. This is used
# to avoid the processing of big files which may take a lot of processing
# time. The value is expressed in bytes, by default 128 MB.
analysis_size_limit = 104857600

# Enable or disable DNS lookups.
resolve_dns = on

# Enable PCAP sorting, needed for the connection content view in the web interface.
sort_pcap = on

[database]
# Specify the database connection string.
# NOTE: If you are using a custom database (different from sqlite), you have to
# use utf-8 encoding when issuing the SQL database creation statement.
# Examples, see documentation for more:
# sqlite:///foo.db
# postgresql://foo:bar@localhost:5432/mydatabase
# mysql://foo:bar@localhost/mydatabase
# If empty, defaults to a SQLite3 database at $CWD/cuckoo.db.
connection = 

# Database connection timeout in seconds.
# If empty, default is set to 60 seconds.
timeout = 60

[timeouts]
# Set the default analysis timeout expressed in seconds. This value will be
# used to define after how many seconds the analysis will terminate unless
# otherwise specified at submission.
default = 120

# Set the critical timeout expressed in (relative!) seconds. It will be added
# to the default timeout above and after this timeout is hit
# Cuckoo will consider the analysis failed and it will shutdown the machine
# no matter what. When this happens the analysis results will most likely
# be lost.
critical = 60

# Maximum time to wait for virtual machine status change. For example when
# shutting down a vm. Default is 60 seconds.
vm_state = 300

[remotecontrol]
# Enable for remote control of analysis machines inside the web interface.
enabled = off

# Set host of the running guacd service.
guacd_host = {{ cuckoo.remotecontrol.guacd_host }}

# Set port of the running guacd service.
guacd_port = {{ cuckoo.remotecontrol.guacd_port }}

**Comment by doomedraven** [comment]
RTFM ;) `cuckoo --help`

**Comment by manzy90** [comment]
@doomedraven ?

**Comment by doomedraven** [comment]
run that

**Comment by manzy90** [comment]
I did

**Comment by doomedraven** [comment]
you have `rooter` there, so you need to do `cuckoo -d rooter --help` for more help, check the documentation it all there

**Comment by manzy90** [comment]
also did this before
sudo cuckoo rooter
2018-10-05 22:53:34,032 [cuckoo] INFO: Starting Cuckoo Rooter (group=cuckoo)!
The group ('cuckoo') does not exist. Please define the group / user through which Cuckoo will connect to the rooter, e.g., 'cuckoo rooter -g myuser'.

**Comment by doomedraven** [comment]
`--help` and documentation, solves all this issues, you need to have it running in parallel with cuckoo to be able to use cuckoo

**Comment by manzy90** [comment]
ok I just found it odd that it was working fine before, i must have stuffed something up :(

**Comment by doomedraven** [comment]
looks like :P if that was working

**Comment by manzy90** [comment]
Thanks so how can I solve this issue?

**Comment by doomedraven** [comment]
reading the documentation is good start!
+ reading output errors is a good clue

```
2018-10-05 22:53:34,032 [cuckoo] INFO: Starting Cuckoo Rooter (group=cuckoo)!
The group ('cuckoo') does not exist. Please define the group / user through which Cuckoo will connect to the rooter, e.g., 'cuckoo rooter -g myuser'.
```

**Comment by manzy90** [comment]
Sorry i tried this 
sudo cuckoo -g root

**Comment by doomedraven** [comment]
never run cuckoo under root, that will give you a headache on long run and that is bad opsec

**Comment by manzy90** [comment]
Ok thanks for that tip..

**Comment by doomedraven** [comment]
as documentation says create user cuckoo, and run everything under user cukoo, thats all :)

**Comment by manzy90** [comment]
ok cool

**Comment by manzy90** [comment]
So how can i solve this issue please?

**Comment by doomedraven** [comment]
__READ THE MANUAL!__

**Comment by manzy90** [comment]
do you think I might have something in the config file that is wrong?

**Comment by doomedraven** [comment]
did you read the manual?

**Comment by manzy90** [comment]
Yes I did, are you here to help or make fun of people when they fall into problems?

**Comment by doomedraven** [comment]
lol, to help but too much people complain without even check documentation, 

which is the problem to start `cuckoo -d rooter -g cuckoo` or your user, thats all i told you

**Comment by manzy90** [comment]
Thanks I am new to Ubuntu and havent used cuckoo before, wanted to give it a try was happy that i got it running. As it is an amazing tool. I then fell into this problem.

My user is Cuckoo,
When i try to run cuckoo -d rooter -g cuckoo i get this again
sudo cuckoo -d rooter -g cuckoo
2018-10-05 23:14:59,622 [cuckoo] INFO: Starting Cuckoo Rooter (group=cuckoo)!
The group ('cuckoo') does not exist. Please define the group / user through which Cuckoo will connect to the rooter, e.g., 'cuckoo rooter -g myuser'.

**Comment by doomedraven** [comment]
as i told before why you don't read the output you have the command to run
`to the rooter, e.g., 'cuckoo rooter -g myuser'.`
in your case to the rooter, e.g., `cuckoo rooter -g Cuckoo`

and again as i told `--help` is helping

**Comment by manzy90** [comment]
Your not helping me at all, i followed the instructions on the website and also this link

**Comment by doomedraven** [comment]
lol i can't help without having details ROFL

**Comment by manzy90** [comment]
https://medium.com/@warunikaamali/cuckoo-sandbox-installation-guide-d7a09bd4ee1f

**Comment by manzy90** [comment]
what details do you need?????

**Comment by doomedraven** [comment]
that which im asking, and here is your issues you didn't read the documenattion, 
as cuckoo now is 2.0.6, that  link is `Cuckoo Sandbox 2.0 rc-1` a lot changed since that, so if you don't like my help, no problem, good luck chao

**Comment by manzy90** [comment]
Next time say that from the start that you can't or don't want to help. I posted here seeking help not for someone to make fun of me for falling into a problem

**Comment by doomedraven** [comment]
lol i didn't do fun of that, how the hell you imagine someone can help you without details? i pointed you to documentation, to your error, you have installation problem, i asked how you installed to point to to correct way, but well if all that is `fun` for you, lol i don't give a fuck dude

**Comment by manzy90** [comment]
Look man i respect that you are an expert in this and need help please, as i mentioned I am new to this. I just need help I tried alot of things before posting I did go to that documentation https://cuckoo.sh/docs/usage/rooter.html. 
When I tried to enter 'cuckoo rooter -g myuser' I got this

2018-10-05 23:48:10,145 [cuckoo] INFO: Starting Cuckoo Rooter (group=Cuckoo)!

As you mentioned I need to leave it open, so i did that then tried to run 'sudo cuckoo' got the same error.

**Comment by doomedraven** [comment]
__then don't be harsh__, people ask details to help, we have a lot of work, and details save time... but for some as you is fun then awesome lol

you need to use `-g` on all  cuckoo services to set them all correct user

**Comment by doomedraven** [comment]
cuckoo -d -g Cuckoo rooter
cuckoo -d -g Cuckoo cuckoo
etc

**Comment by manzy90** [comment]
Fair enough apologies for that

**Comment by manzy90** [comment]
I dont know where i have stuffed up :( I am so confused.

**Comment by doomedraven** [comment]
if you have this in config `rooter = {{ cuckoo.cuckoo.rooter }}` as `git clone` without `python setup.py install` is bad installation, remove everything, and install from `sudo pip install cuckoo`

**Comment by doomedraven** [comment]
do this https://cuckoo.sh/docs/installation/host/installation.html

**Comment by manzy90** [comment]
ok thanks will do that

**Comment by doomedraven** [comment]
i hope now you understand the importane of all that stupid questions, nobody likes when people ask them, `did you try to reboot` lol

**Comment by manzy90** [comment]
Yes of course, I understand.

**Comment by manzy90** [comment]
@doomedraven  It worked thank you very much!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by doomedraven** [comment]
you are welcome
1. close issue
2. gain patience even with stupid questions, or provide as much info as you can even if that sound stupid, initially to speedup issue solving and saving time to the rest
3. enjoy cuckoo ;)
